sdk/internal/logger/logger.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package logger
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+// Listener is the function signature invoked when writing log entries.
+// A Listener is required to perform its own synchronization if it's
+// expected to be called from multiple Go routines.
+type Listener func(azcore.LogClassification, string)
```

Comment:
Copy the `azcore.LogClassification` type over to the internal module. Then in `azcore` type alias the `LogClassification` type over there if if you need it.

Reply:
All the below references to the `azcore.LogClassification` will then be to the one referenced here to remove the dependency on `azcore`.